### PR TITLE
Docker changes

### LINKED ISSUE #1: Add a tag to auto pull when compose up
# Overview
Right now, we need to remember to always docker pull before doing any work, with this option enabled, we won't need that anymore. It will auto update.
# Task
Add the ```pull_policy: always``` in our ```docker-compose.yml```file to always pull the latest image from our repo.
# References
https://docs.docker.com/compose/compose-file/#pull_policy

### LINKED ISSUE #6: Change base image of our Docker container
# Overview
Right now, we're using the Ubuntu 20.04 image and manually installing ROS and the rest of our dependencies. We should use the ```noetic-ros-base-focal``` that already comes with more sane defaults for our use case.
# Tasks
Edit the Dockefile to use the ```noetic-ros-base-focal``` image and adapt the rest of the file

### LINKED ISSUE #7: Add a tag to auto pull when compose up
# Overview
Right now, we need to remember to always docker pull before doing any work, with this option enabled, we won't need that anymore. It will auto update.
# Task
Add the ```pull_policy: always``` in our ```docker-compose.yml```file to always pull the latest image from our repo.
# References
https://docs.docker.com/compose/compose-file/#pull_policy

### PR DESCRIPTION
**Pull request by Sobrenho**
Don't forget to change path to container in docker-compose.yml after merge. Dockerfile might still have some redundant installs. I'll continue to do some testing.

### CODE REVIEW

**Comment by matheusrod98** [comment]
Delete the ROS installation from the Dockerfile to be merged. It's not needed anymore since it already comes installed in the ROS image.

File: docker-compose.yml [hunk 1]
```diff
@@ -5,10 +5,11 @@ services:
     container_name: auv_ws
     image: ghcr.io/ufrj-nautilus/auv_ws:master
     privileged: true
+    pull_policy: always
     environment:
       - DISPLAY=${DISPLAY} 
     volumes:
-      - ./../../../catkin_ws:/root/catkin_ws
+      - ./../catkin_ws:/root/catkin_ws
```
Comment (matheusrod98) [inline docker-compose.yml]: Change this to mount the catkin_ws/src in the container.
Probably this:
```
- ./../src/:/root/catkin_ws/src
```

We don't need build and devel outside the container anyway.

**Review by matheusrod98 — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -16,9 +15,7 @@ RUN apt update && DEBIAN_FRONTEND=noninteractive apt install ros-noetic-desktop-
 RUN echo "source /opt/ros/noetic/setup.bash" >> /root/.bashrc
 RUN echo "source /usr/share/gazebo/setup.sh" >> /root/.bashrc
 RUN source /root/.bashrc
-RUN apt update && apt install python3-rosdep python3-rosinstall python3-rosinstall-generator python3-wstool python3-serial build-essential -y
-RUN rosdep init
-RUN rosdep update
+RUN apt update && apt install python3-rosinstall-generator python3-wstool python3-serial  -y
```
Comment (matheusrod98) [inline Dockerfile]: This was necessary?

Comment (Sobrenho) [reply Dockerfile]: These ones already come installed in the ROS image. 

I think we can also remove gnupg on line 8 since the ROS image has gnupg2.

Comment (Sobrenho) [reply Dockerfile]: https://hub.docker.com/layers/library/ros/noetic-ros-base-focal/images/sha256-2bb6821186d86445b5f0b2da7a4ccc36dbc26707ae18b8c8b52184baf0c5fd11?context=explore

Comment (matheusrod98) [reply Dockerfile]: Yeah, if possible, remove everything that already comes with the ROS image.

Comment (matheusrod98) [reply Dockerfile]: Let's try to have a minimal Dockerfile, using most of the default ROS image when possible.

File: docker-compose.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ services:
     environment:
       - DISPLAY=${DISPLAY} 
     volumes:
-      - ./../catkin_ws:/root/catkin_ws
+      - ./../catkin_ws:/root/catkin_ws/src
```
Comment (matheusrod98) [inline docker-compose.yml]: Just mount the src directory, there is no need to mount entire catkin_ws.
```
- ./../catkin_ws/src:/root/catkin_ws/src
```

**Review by matheusrod98 — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -1,5 +1,4 @@
-# Uses the Ubuntu 20.04 LTS (Focal Fossa) as the base image.
-FROM ubuntu:20.04
+FROM ros:noetic-ros-base-focal
 
 # Shell to be used during the build process and the container's default.
```
Comment (matheusrod98) [inline Dockerfile]: Remove lines 3 and 4. There is no need to set the default shell, the ROS image already does this.

File: Dockerfile [hunk 5]
```diff
@@ -9,16 +8,14 @@ RUN apt update && apt upgrade -y
 RUN apt update && apt install -y gnupg wget lsb-release vim tmux libglvnd0 libglvnd-dev x11-apps mesa-utils python3-pip htop
```
Comment (matheusrod98) [inline Dockerfile]: Only install tmux, vim, htop, x11-apps, mesa-utils and python3-pip
Delete line 7

File: Dockerfile [hunk 6]
```diff
@@ -9,16 +8,14 @@ RUN apt update && apt upgrade -y
...
 RUN yes | pip3 install pynput pymap3d
 
-# Install ROS noetic
-RUN sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+# Install ROS noetic additional dependencies
```
Comment (matheusrod98) [inline Dockerfile]: Change this to:
```
# Settings ROS development environment.
```

File: Dockerfile [hunk 7]
```diff
@@ -9,16 +8,14 @@ RUN apt update && apt upgrade -y
...
 RUN wget -q https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc -O- | apt-key add -
```
Comment (matheusrod98) [inline Dockerfile]: The gpg key of the repo it's already bundled in the image, delete the line 12

**Review by matheusrod98 — CHANGES_REQUESTED** [review]

**Comment by Sobrenho** [comment]
Do I add meshlab and dearpygui ?

**Comment by matheusrod98** [comment]
> Do I add meshlab and dearpygui ?

Yes, please. I forgot both of them.

File: Dockerfile [hunk 8]
```diff
@@ -1,28 +1,20 @@
-# Uses the Ubuntu 20.04 LTS (Focal Fossa) as the base image.
-FROM ubuntu:20.04
-
-# Shell to be used during the build process and the container's default.
-SHELL ["/bin/bash", "-c"]
+FROM ros:noetic-ros-base-focal
 
 # Update the system and install some essential packages.
```
Comment (matheusrod98) [inline Dockerfile]: Update comment, we don't ```apt update ; apt upgrade``` anymore.

File: Dockerfile [hunk 9]
```diff
@@ -1,28 +1,20 @@
...
-RUN wget -q https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc -O- | apt-key add -
+# Settings ROS develepment environment
 RUN apt update && DEBIAN_FRONTEND=noninteractive apt install ros-noetic-desktop-full ros-noetic-rtabmap-ros ros-noetic-robot-localization liburdfdom-tools -y
 RUN echo "source /opt/ros/noetic/setup.bash" >> /root/.bashrc
 RUN echo "source /usr/share/gazebo/setup.sh" >> /root/.bashrc
-RUN source /root/.bashrc
-RUN apt update && apt install python3-rosdep python3-rosinstall python3-rosinstall-generator python3-wstool python3-serial build-essential -y
-RUN rosdep init
-RUN rosdep update
+RUN apt update && apt install python3-rosinstall-generator python3-wstool python3-serial meshlab -y
```
Comment (matheusrod98) [inline Dockerfile]: Remove everything but ```meshlab``` and ```python3-serial```.

File: Dockerfile [hunk 10]
```diff
@@ -1,28 +1,20 @@
...
-RUN apt update && apt upgrade -y
-RUN apt update && apt install -y gnupg wget lsb-release vim tmux libglvnd0 libglvnd-dev x11-apps mesa-utils python3-pip htop
-RUN yes | pip3 install pynput pymap3d
+RUN apt update && apt install -y vim tmux  x11-apps mesa-utils python3-pip htop
+RUN yes | pip3 install pynput pymap3d dearpygui
 
-# Install ROS noetic
-RUN sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
-RUN wget -q https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc -O- | apt-key add -
+# Settings ROS develepment environment
```
Comment (matheusrod98) [inline Dockerfile]: Replace with ```Installing Ubuntu dependencies```

**Review by matheusrod98 — CHANGES_REQUESTED** [review]

**Comment by matheusrod98** [comment]
Ok, it seems fine for me. I'm gonna test it before merge and close this PR.

**Review by matheusrod98 — APPROVED** [review]

**Pull request merged**